pynio/service.py

```diff
@@ -38,6 +38,13 @@ def connect(self, blk1, blk2):
             if blk['name'] == blk1.name:
                 connection = blk
                 break
+        for blk in execution:
+            if blk['name'] == blk2.name:
+                break
+        else:
+            # block2 doesn't exist, so add it
+            execution.append({'name': blk2.name, 'receivers': []})
```

Comment:
I'm pretty sure this isn't necessary. Can you verify that nio actually requires the receiving block to be added to the service?